src/contexts/OracleDataContext.tsx

```diff
@@ -186,17 +185,59 @@ export function OracleDataProvider({ children }: { children: ReactNode }) {
 
   useEffect(() => {
     // its important this client only gets initialized once
-    Client([oraclesService, ...oracleEthersServices], {
+    Client([...oraclesServices, ...oracleEthersServices], {
       requests: (requests) => dispatch({ type: "requests", data: requests }),
       assertions: (assertions) =>
         dispatch({ type: "assertions", data: assertions }),
       errors: setErrors,
     });
   }, []);
 
-  errors.forEach((error, i) => {
-    if (error) console.warn("Oracle data error", i, errors);
-  });
+  useEffect(() => {
+    errors.forEach((error, i) => {
+      if (error == undefined) return;
+      // index of service must align with order configs are passed into client
+      const serviceConfig = serviceConfigs[i];
+      // this error is coming from ether provider queries, this would mean fallback is broken
+      if (serviceConfig?.source !== "gql") {
+        console.warn({ serviceConfig, error });
+        addErrorMessage({
+          text: "Currently unable to fetch all data, check back later",
+          link: {
+            text: "Use the Legacy Dapp instead",
+            href: "https://legacy.oracle.uma.xyz",
+          },
+        });
+        return;
+      }
+      const web3Fallback =
+        oracleEthersApis?.[serviceConfig.type]?.[serviceConfig.chainId];
+      if (web3Fallback && web3Fallback.queryLatestRequests) {
+        const web3Config = config.providers.find(
+          (c) =>
+            c.type == serviceConfig.type && c.chainId == serviceConfig.chainId
+        );
+        web3Fallback.queryLatestRequests(
+          web3Config?.blockHistoryLimit ?? 100000
+        );
+        // if we reach here, theres a subgraph thats not working, but we can still fetch limited  history through provider
+        addErrorMessage({
+          text: "The Graph is experiencing downtime, site may be slower than normal",
+        });
+      } else {
+        // if we reach here, we dont have a fallback for a specific subgraph, technically this shouldnt ever happen though
+        // if the app is configured correctly
+        console.warn({ serviceConfig, error });
+        addErrorMessage({
+          text: "Currently unable to fetch all data, check back later",
```

Comment:
I tried testing this by blocking api.thegraph.com in my browser's network tab. I also tried the way you suggested by changing the urls in my env to be wrong.

I'm getting the "Currently unable to fetch all data, check back later" error in my error banner, instead of the "The Graph is experiencing downtime, site may be slower than normal" one I expected.

The actual fetching seems to work well though 👍